pkg/controller/linstornodeset/linstornodeset_controller.go

```diff
@@ -489,9 +482,7 @@ func (r *ReconcileLinstorNodeSet) reconcileStoragePoolsOnNode(sat *piraeusv1alph
 		}).Debug("NS reconcileStoragePoolsOnNode: found storage pool")
 
 		// Guard against empty statuses.
-		if status == nil || status.Name != "" {
```

Comment:
Have you checked whether/how this might occur? One assumes that there was some reason for it in the first place.

Reply:
Its a bit strange, because in case the `status == nil` condition holds, this will lead to a nil dereference.
Also status is created in line 485 above, which is ensures that status is never nil.